File: techsec-frontend/src/app/api/models/response-product-get-all.ts
import { ResponseCategoryGetAll } from './response-category-get-all';

export interface ResponseProductGetAll {
  id: number;
  name: string;
  category?: ResponseCategoryGetAll;
  description?: string;
  specifications?: string;
  price: number;
  stock: number;
  minStockAlert: number;
  imageUrl?: string;
}

File: apitechsec/src/main/java/com/epiis/apitechsec/dto/response/ResponseProductGetAll.java
package com.epiis.apitechsec.dto.response;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

import com.epiis.apitechsec.generic.ResponseGeneric;
import lombok.Getter;
import lombok.Setter;

public class ResponseProductGetAll extends ResponseGeneric {
	public List<ItemProduct> listProduct = new ArrayList<>();

	@Getter
	@Setter
	public static class ItemProduct {
		public Long id;
		public String name;
		public Long idCategory;
		public String categoryName;
		public String description;
		public String specifications;
		public BigDecimal price;
		public Integer stock;
		public Integer minStockAlert;
	}
}
